### ISSUE TITLE
grub menu does not appear on serial console

### ISSUE THREAD

**Issue by mgerdts** [joyent/linux-live#11]
I have been unable to convince the grub menu to appear anywhere other than the graphical console.  This is not great for remote operations.

I've tried various incantations similar to the following with no progress:

```
diff --git a/tools/debian-live b/tools/debian-live
index a634a17..32565af 100755
--- a/tools/debian-live
+++ b/tools/debian-live
@@ -356,11 +356,15 @@ function prepare_archive_bits {
        cat <<EOF >$scratch/grub.cfg
 search --set=root --file /JOYENT_DEBIAN_LIVE

-insmod all_video
+#insmod all_video

 set default="0"
 set timeout=30

+serial --speed=115200 --unit=0 --word=8 --parity=no --stop=1
+terminal_input serial
+terminal_output serial
+
 menuentry "Joyent Debian Live $release without DHCP ttyS0" {
     linux /vmlinuz boot=live console=ttyS0
     initrd /initrd
```

In the `terminal_*` lines, I've tried `console serial` and `--append serial`.  Once grub starts the kernel, the kernel's messages are seen on the serial port.

**Comment by jserviceorg** [comment]
i dont know if it makes any difference, but i was using something like this in the post with success (bios configuration maybe needed as well)

console=tty0 console=ttyS0,115200n8

**Comment by mgerdts** [comment]
@jserviceorg I think you are referring to the kernel command line.  That part is working just fine.  The part that's not working is the display of grub menu on the serial console.  Or do I misunderstand you?

**Comment by jserviceorg** [comment]
nope, you understood me correctly. i missed that its just about the menu itself. did you try:

/etc/default/grub

+GRUB_TERMINAL="console serial"


https://askubuntu.com/questions/924913/how-to-get-to-the-grub-menu-at-boot-time-using-serial-console/1110209

**Comment by mgerdts** [comment]
`/etc/default/grub` is used by `grub2-mkconfig` to generate a `grub.cfg` file.  Since I need to generate a grub configuration that will reside on iso and usb images, the standard grub tooling that does device probing, etc., is ill suited to the task at hand.  [`debian-live`](https://github.com/joyent/linux-live/blob/linuxcn/tools/debian-live) creates a minimal `grub.cfg` without `grub2-mkconfig`.

That being said, on another machine I did just as you suggested, then ran `grub2-mkconfig` and noticed that its output had the `serial` and `terminal_*` lines that I tried - that's how I figured out what commands to try in the `grub.cfg` because there is almost no documentation of how to configure grub2 with using `/etc/default/grub`.

**Comment by aqw** [comment]
For my Linux systems, I have the following in a file under `/etc/default/grub.d`

```
GRUB_CMDLINE_LINUX_DEFAULT="nomodeset console=tty0 console=ttyS1,115200n8"
GRUB_TERMINAL=serial
GRUB_SERIAL_COMMAND="serial --speed=115200 --unit=1 --word=8 --parity=no --stop=1"
```
